Concurrent createrepo runs can race and cause repo metadata corruption

Probably caused by non-atomic updates of repo metadata.

```
Traceback (most recent call last):
File "/usr/bin/planex-cache", line 9, in <module>
load_entry_point('planex==0.6.0', 'console_scripts', 'planex-cache')()
File "/usr/lib/python2.7/site-packages/planex-0.6.0-py2.7.egg/planex/cache.py", line 239, in
_main
main(sys.argv[1:])
File "/usr/lib/python2.7/site-packages/planex-0.6.0-py2.7.egg/planex/cache.py", line 217, in
main
pkg_hash = get_srpm_hash(srpm, yumbase, mock_config)
File "/usr/lib/python2.7/site-packages/planex-0.6.0-py2.7.egg/planex/cache.py", line 156, in
get_srpm_hash
pkgs = yumbase.pkgSack.returnNewestByNameArch(patterns=[req])
File "/usr/lib/python2.7/site-packages/yum/packageSack.py", line 520, in
returnNewestByNameArch
pkgs = calr("returnNewestByNameArch", naTup, patterns, ignore_case)
File "/usr/lib/python2.7/site-packages/yum/packageSack.py", line 595, in
_computeAggregateListResult
sackResult = apply(method, args)
File "/usr/lib/python2.7/site-packages/yum/sqlitesack.py", line 46, in newFunc
return func(*args, **kwargs)
File "/usr/lib/python2.7/site-packages/yum/sqlitesack.py", line 1472, in
returnNewestByNameArch
ignore_case)
File "/usr/lib/python2.7/site-packages/yum/packageSack.py", line 951, in
returnNewestByNameArch
ignore_case=ignore_case)
File "/usr/lib/python2.7/site-packages/yum/sqlitesack.py", line 1669, in returnPackages
pkgobjlist = self._buildPkgObjList(repoid, patterns, ignore_case)
File "/usr/lib/python2.7/site-packages/yum/sqlitesack.py", line 46, in newFunc
return func(*args, **kwargs)
File "/usr/lib/python2.7/site-packages/yum/sqlitesack.py", line 1628, in _buildPkgObjList
return self.searchNames(patterns)
File "/usr/lib/python2.7/site-packages/yum/sqlitesack.py", line 57, in newFunc
raise Errors.RepoError, str(e)
yum.Errors.RepoError: database disk image is malformed
/usr/share/planex/Makefile.rules:69: recipe for target
'_build/RPMS/x86_64/ocaml402-ocaml-text-0.7.1-2.el6.x86_64.rpm' failed
make: *** [_build/RPMS/x86_64/ocaml402-ocaml-text-0.7.1-2.el6.x86_64.rpm] Error 1
make: *** Waiting for unfinished jobs....
```

--- a/planex/cache.py
+++ b/planex/cache.py
@@ -69,7 +69,7 @@ def setup_yumbase(yumbase, loopback_repo):
     yumbase.repos.disableRepo('*')
     yumbase.repos.enableRepo(loopback_repo)
 
-    yumbase.setCacheDir(force=True, reuse=True)
+    yumbase.setCacheDir(force=True, reuse=False)
     yumbase.repos.populateSack(cacheonly=True)
 
 

